src/Symfony/Component/Cache/Adapter/RedisTagAwareAdapter.php

```diff
@@ -305,4 +306,185 @@ private function getRedisEvictionPolicy(): string
+        $getTagsLua = <<<'EOLUA'
+            redis.replicate_commands()
+            local cursor = ARGV[1]
+            local prefix = ARGV[2]
+            local tagPrefix = string.gsub(KEYS[1], prefix, "")
+            return redis.call('SCAN', cursor, 'COUNT', 5000, 'MATCH', '*' .. tagPrefix .. '*', 'TYPE', 'set')
+        EOLUA;
+        $cursor = null;
+        do {
+            $results = $this->pipeline(function () use ($getTagsLua, $cursor, $prefix, $tagsPrefix) {
+                yield 'eval' => [$getTagsLua, [$tagsPrefix, $cursor, $prefix], 1];
+            });
+
+            $setKeys = $results->valid() ? iterator_to_array($results) : [];
+            [$cursor, $ids] = $setKeys[$tagsPrefix] ?? [null, null];
+            // merge the fetched ids together
+            $tagKeys = array_merge($tagKeys, $ids);
+        } while ($cursor = (int) $cursor);
+
+        return $tagKeys;
+    }
+
+    /**
+     * Checks all tags in the cache for orphaned items and creates a "report" array.
+     *
+     * By default, only completely orphaned tag keys are reported. If
+     * compressMode is enabled the report will include all tag keys
+     * that have any orphaned references to cache items
+     *
+     * @TODO Verify the LUA scripts are redis-cluster safe.
+     * @TODO Is there anything that can be done to reduce memory footprint?
+     *
+     * @return array{tagKeys: string[], orphanedTagKeys: string[], orphanedTagReferenceKeys?: array<string, string[]>}
+     *                                                                                                                 tagKeys: List of all tags in the cache.
+     *                                                                                                                 orphanedTagKeys: List of tags that only reference orphaned cache items.
+     *                                                                                                                 orphanedTagReferenceKeys: List of all orphaned cache item references per tag.
+     *                                                                                                                 Keyed by tag, value is the list of orphaned cache item keys.
+     */
+    private function getOrphanedTagsStats(bool $compressMode = false): array
+    {
+        $prefix = $this->getPrefix();
+        $tagKeys = $this->getAllTagKeys();
+
+        // lua for fetching all entries/content from a SET
+        $getSetContentLua = <<<'EOLUA'
+            redis.replicate_commands()
+            local cursor = ARGV[1]
+            return redis.call('SSCAN', KEYS[1], cursor, 'COUNT', 5000)
+        EOLUA;
+
+        $orphanedTagReferenceKeys = [];
+        $orphanedTagKeys = [];
+        // Iterate over each tag and check if its entries reference orphaned
+        // cache items.
+        foreach ($tagKeys as $tagKey) {
+            $tagKey = substr($tagKey, \strlen($prefix));
+            $cursor = null;
```

Comment:
This should be 0, null leads to something like: `ERR invalid cursor script: 5880028e0d564320af1c1aabb595c9985f5a1a04, on @user_script:5.`
This is used in SCAN which expects 0 to start and returns 0 when the iteration is finished: https://redis.io/docs/latest/commands/scan/ 
Unfortunately I can't find older docs and have to assume this is backward compatible.